roachtest: update pg_regress diff

yuzefovich opened the pull request:
This commit updates the diff based on recent line number changes. The good news is that the internal errors that print the stack trace should be fixed shortly (PR is already out).

I also spent some time trying to understand what would it take to make `pg_regress` a part of the local roachtest run in CI (so that we don't get these nightly failures and, instead, update the diff as part of the change that triggers the diff), but it seems to be a no-go for two reasons:
- we generally want to make CI run faster, and pg_regress takes on the order of 10 minutes on a gceworker which would make it the slowest roachtest that is part of CI I think
- we run local roachtests in CI inside of the docker container, which at least currently doesn't have `sudo` command to install things like postgres needed for this roachtest.

So I think for now we'll need to either update the diff based on the nightly failure or proactively manually kick of a run of pg_regress when we know that the diff might need to be changed.

Fixes: #119626.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/119645)
<!-- Reviewable:end -->

rytaft commented on pkg/cmd/roachtest/testdata/regression.diffs:
```diff
@@ -162375,10 +162376,10 @@ diff -U3 --label=/mnt/data1/postgres/src/test/regress/expected/collate.windows.w
 +)
 +^
-+INSERT INTO collate_test1 VALUES (1, 'abc'), (2, '�bc'), (3, 'bbc'), (4, 'ABC');
++INSERT INTO collate_test1 VALUES (1, 'abc'), (2, '�bc'), (3, 'bbc'), (4, 'ABC');
```
what changed to cause this diff?

yuzefovich replied:
I'm not quite sure - perhaps the last time we simply downloaded artifacts from the failed nightly run and this time I manually kicked of a run. The difference seems to be with unicode characters. I reverted these changes.

rytaft reviewed (approved):
LGTM, just one question below

yuzefovich committed:
roachtest: update pg_regress diff

This commit updates the diff based on recent line number changes. The
good news is that the internal errors that print the stack trace should
be fixed shortly (PR is already out).

I also spent some time trying to understand what would it take to make
`pg_regress` a part of the local roachtest run in CI (so that we don't
get these nightly failures and, instead, update the diff as part of the
change that triggers the diff), but it seems to be a no-go for two
reasons:
- we generally want to make CI run faster, and pg_regress takes on the
order of 10 minutes on a gceworker which would make it the slowest
roachtest that is part of CI I think
- we run local roachtests in CI inside of the docker container, which at
least currently doesn't have `sudo` command to install things like
postgres needed for this roachtest.

So I think for now we'll need to either update the diff based on the
nightly failure or proactively manually kick of a run of pg_regress when
we know that the diff might need to be changed.

Release note: None

yuzefovich reviewed (commented):
TFTR!

bors r+
